src/ejabberd_cache_mnesia.erl

```diff
@@ -0,0 +1,175 @@
+%%%-------------------------------------------------------------------
+%%% File    : ejabberd_cache_mnesia.erl
+%%% Author  : Devin Sills <dsills@skillz.com>
+%%% Created : 18 Jan 2022 by Devin Sills <dsills@skillz.com>
+%%%
+%%%----------------------------------------------------------------------
+
+-module(ejabberd_cache_mnesia).
+
+-behaviour(gen_server).
+
+%% External exports
+-export([start_link/2, get_proc_name/1]).
+
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2, terminate/3, handle_info/3]).
+
+-include("logger.hrl").
+
+start_link(CacheName, MaxCacheSize) ->
+  %% sql cache table
+  mnesia:create_table(get_cache_table_name(CacheName), [{attributes, [ key, value ]}, {type, set}, {ram_copies, [node()]}, {local_content, false}]),
+
+  case gen_server:start_link({local, get_proc_name(CacheName)}, ?MODULE, [CacheName, MaxCacheSize], []) of
+    {ok, Pid} ->
+      {ok, Pid};
+    {error, {already_started, Pid}} ->
+      {ok, Pid};
+    ignore -> ignore;
+    {error, Error} -> {error, Error};
+    Other -> Other
+  end
+.
+
+init([CacheName, MaxCacheSize]) ->
+  FullCacheName = get_cache_table_name(CacheName),
+  {ok, {FullCacheName, MaxCacheSize, get_size(FullCacheName, MaxCacheSize)}}.
```

Comment:
nit for consistency looks like mostly putting the '.' on the next line in the rest of this file